homeassistant/components/sensor/__init__.py

```diff
@@ -95,21 +96,43 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return await component.async_unload_entry(entry)
 
 
+@dataclass
+class SensorEntityDescription(EntityDescription):
+    """An class that describes sensor entities."""
+
+    state_class: str | None = None
+    last_reset: datetime | None = None
+
+
 class SensorEntity(Entity):
     """Base class for sensor entities."""
 
-    _attr_state_class: str | None = None
-    _attr_last_reset: datetime | None = None
+    _attr_entity_description: SensorEntityDescription
+    _attr_state_class: str | None
+    _attr_last_reset: datetime | None
+
+    @property
+    def entity_description(self) -> SensorEntityDescription:
```

Comment:
Is it even necessary to make this a property? Why not use `self._attr_entity_description` directly? It would skip 1 function hop and so be faster.

Reply:
I mainly added it for consistency. As every `_attr_*` has a matching property

Reply:
I think that it's fine to omit it. Those other properties exist for legacy reasons and we probably wouldn't add them today.